mcu-util/src/orb/main_board.rs

```diff
@@ -328,9 +328,10 @@ impl MainBoardInfo {
     }
 
     /// Fetches `MainBoardInfo` from the main board
+    /// doesn't fail, but lazily fetches as much info as it could
     /// on timeout, returns the info that was fetched so far
-    async fn build(mut self, main_board: &mut MainBoard) -> Result<Self> {
-        main_board
+    async fn build(mut self, main_board: &mut MainBoard) -> Self {
```

Comment:
Instead, make this a `Result<Self, Self>`. The err variant is when not all data was fetched.